docs/usage/1-route-handlers.md

```diff
@@ -0,0 +1,350 @@
+
+* `path` (**required**) - a path string, with or without [path parameters](#path-parameters).
+* `http_method` (**required**) - a member of the `HttpMethod` enum or a list of members, e.g. `HttpMethod.GET`
+  or `[HttpMethod.Patch, HttpMethod.Put]`.
+* `status_code`: the status code for a success response. If not specified, a default value will be used:
+  HTTP_200_OK for _get_, _put_ and _patch_, HTTP_204_NO_CONTENT for _delete_, and HTTP_201_CREATED for _post_. **
+  Important note**: if you specify more than one http method, as in the example above, you must specify the status_code
+  to use for the response.
+* `media_type`: A string or a member of the enum `starlite.MediaType`, which specifies the MIME Media Type for the
+  response. Defaults to `MediaType.JSON`.
+* `response_class`: The response class to use. The value must be either a `Starlette` response class or a class that
+  extends it. Defaults to `starlite.Response`.
+* `response_headers`: A _dataclass_ or _pydantic model_ that describes the response headers. This value is
+  used only in the OpenAPI schema. Defaults to `None`.
+* `dependencies`: A dictionary mapping dependency injection Providers to keys. Defaults to `None`.
+* `include_in_schema`: A boolean flag dictating whether the given route handler will appear in the generated OpenAPI
+  schema. Defaults to `True`.
+* `tags`: a list of openapi-pydantic `Tag` models, which correlate to
+  the [tag specification](https://spec.openapis.org/oas/latest.html#tag-object). Defaults to `None`.
+* `summary`: Text used for the route's schema _summary_ section. Defaults to `None`.
+* `description`: Text used for the route's schema _description_ section. Defaults to `None`.
+* `operation_id`: An identifier used for the route's schema _operationId_. Defaults to `None`. Because this is a
+  required OpenAPI value, if it's not provided by the user, the function/method's name will be used.
+* `deprecated`: A boolean dictating whether this route should be marked as deprecated in the OpenAPI schema. Defaults
+  to `False`.
+* `raises`: A list of exception classes extending from `starlite.HttpException`. This list should describe all
+  exceptions raised within the route handler's function/method. The StarLite `ValidationException` will be added
+  automatically for the schema if any validation is involved.
+
+## Semantic Handler Decorators
+
+StarLite also includes the following decorators, which as their names suggest already pre-set the `http_method` kwarg:
+
+* `delete`
+* `get`
+* `patch`
+* `post`
+* `put`
+
+```python
+from starlite import delete, get, patch, post, put
+
+
+@delete(path="/my-endpoint")
+def my_post_handler():
+    ...
+
+
+@get(path="/my-endpoint")
+def my_get_handler():
+    ...
+
+
+@patch(path="/my-endpoint")
+def my_post_handler():
+    ...
+
+
+@post(path="/my-endpoint")
+def my_post_handler():
+    ...
+
+
+@put(path="/my-endpoint")
+def my_post_handler():
+    ...
```

Comment:
This might be a lack of relevant experience, but I find bulk-update and bulk-delete endpoints to be pretty rare. I might have seen one or two in the wild, but have never actively developed against one, and have definitely never created one. Personally I would find this list of endpoints more natural.

```suggestion
@get(path="/resources")
def list_resources():
    ...

@post(path="/resources")
def create_resource():
    ...

@get(path="/resources/{pk}")
def retrieve_resource(pk: int):
    ...

@put(path="/resources/{pk}")
def update_resource(pk: int):
    ...
    
@patch(path="/resources/{pk}")
def partially_update_resource(pk: int):
    ...

@delete(path="/resources/{pk}")
def delete_resource(pk: int):
    ...
```

Reply:
so, as i explain in the docs - you have to type your path params. Can you amend this comment so all {pk} are {pk:int}?